### PR TITLE
Fix concurrent map writes to oldClients

### PR DESCRIPTION
**Pull request by roboslone**

### CODE REVIEW

**Comment by roboslone** [comment]
@jarcoal could you please take a look at this?

File: race_test.go [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import (
+	"net/http"
+	"testing"
+
+	. "github.com/jarcoal/httpmock"
+)
+
+func TestActivateNonDefaultRace(t *testing.T) {
+	for i := 0; i < 10; i++ {
+		go func() {
```
Comment (maxatome) [inline race_test.go]:
```suggestion
		go ActivateNonDefault(&http.Client{})
```

**Review by maxatome — CHANGES_REQUESTED** [review]
Hi @roboslone,
thanks for your contribution.
Could you squash your commits, please?

**Comment by roboslone** [comment]
@maxatome
Squashed commits & fixed `race_test.go`.

File: transport.go [hunk 2]
```diff
@@ -777,6 +777,9 @@ var InitialTransport = http.DefaultTransport
 // than http.DefaultClient).
 var oldClients = map[*http.Client]http.RoundTripper{}
 
+// oldClientsLock protects oldClients from concurrent writes.
+var oldClientsLock = &sync.Mutex{}
```
Comment (maxatome) [inline transport.go]: Sorry last thing, no need of a pointer here:
```suggestion
var oldClientsLock sync.Mutex
```

Comment (roboslone) [reply transport.go]: No worries :) Fixed.

**Comment by maxatome** [comment]
Thanks!

**Comment by roboslone** [comment]
@maxatome Do I need to do anything for a new version to become available in [pkg.go.dev](https://pkg.go.dev/github.com/jarcoal/httpmock)?

**Comment by maxatome** [comment]
I have to publish a new release. I'll try to do it before next Monday.

**Comment by maxatome** [comment]
https://github.com/jarcoal/httpmock/releases/tag/v1.2.0

**Pull request merged**